config/config.go

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package config
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/logger"
+	"gopkg.in/yaml.v2"
+
+	hzcerror "github.com/hazelcast/hazelcast-commandline-client/errors"
+)
+
+const defaultConfigFilename = "config.yaml"
+const (
+	DefaultClusterAddress = "localhost:5701"
+	DefaultClusterName    = "dev"
+)
+const HZCConfKey = "hzc-config"
+
+func ToContext(ctx context.Context, conf *hazelcast.Config) context.Context {
+	return context.WithValue(ctx, HZCConfKey, conf)
+}
+
+func FromContext(ctx context.Context) *hazelcast.Config {
+	return ctx.Value(HZCConfKey).(*hazelcast.Config)
+}
+
+type SSLConfig struct {
+	ServerName         string
+	InsecureSkipVerify bool
+	CAPath             string
+	CertPath           string
+	KeyPath            string
+	KeyPassword        string
+}
+
+type Config struct {
+	Hazelcast hazelcast.Config
+	SSL       SSLConfig
+}
+
+type PersistentFlags struct {
```

Comment:
The name confuses with actual `PersistentFlag` of `cobra`. Also these are not a flag, they keeps the value of the flags and it makes them a lvalue. Therefore we can name as `RootPFlagValues` or something different.

Reply:
Renamed it as `GlobalFlagValues` at https://github.com/hazelcast/hazelcast-commandline-client/pull/47/commits/e46365e1fa90cd22e9ffaeee4a8a349c1dac33d6